### ISSUE TITLE
Docker Desktop 4.16.0 can't use sudo inside container (nosuid base filesystem?)

### ISSUE THREAD

**Issue by rfay** [docker/for-mac#6675]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: 165EF212-DB99-4FE1-90A4-E516A309B23F/20230112235925

### Expected behavior

sudo inside container should be possible to use

### Actual behavior

```
docker run -it --rm -u 33:33 drud/ddev-webserver:v1.21.4 sudo ls
sudo: effective uid is not 0, is /usr/bin/sudo on a file system with the 'nosuid' option set or an NFS file system without root privileges?
```

### Information
  - macOS Version: 13.1 Ventura
  - Intel chip or Apple chip: Both (and Windows)
  - Docker Desktop Version: 4.16.0

### Steps to reproduce the behavior

```
docker run -it --rm -u 33:33 drud/ddev-webserver:v1.21.4 sudo ls
```

This same thing happens on Docker Desktop for Windows, both from Traditional Windows and WSL2, reported in
* https://github.com/docker/for-win/issues/13161

**Comment by kbluescode** [comment]
Failing on Monterey too.

macOS Version: 12.6.2 Monterey
Chip: Apple M1 Pro
Docker Desktop Version: 4.16.0

After reinstalling Docker Desktop 4.15.0 the same container ran properly again.

**Comment by ericvn** [comment]
We use `su-exec 0:0 [command] in our container at it fails with 4.16 as well. Has been working fine for a few years up to this point. Going back to 4.15 fixes the issue. I did try both gRPC and VirtioFS on 4.16 and it failed with both.

**Comment by gamerson** [comment]
I'm seeing this error
```
sudo: The "no new privileges" flag is set, which prevents sudo from running as root.
sudo: If sudo is running in a container, you may need to adjust the container configuration to disable the flag.
```
On the exact container that worked on Monterey + 4.15

**Comment by ericvn** [comment]
Also a pointer to https://github.com/kubernetes/minikube/issues/15634.

**Comment by back-2-95** [comment]
Yup, all our images (both building and running existing images) fail with this.

**Comment by mat007** [comment]
Thank you for reporting this issue.
We are investigating it, and will very likely release a 4.16.1 version of Docker Desktop when we have a fix.

**Comment by mat007** [comment]
Closing this issue because a fix has been released in Docker Desktop 4.16.1. See the [release notes](https://docs.docker.com/desktop/release-notes/) for more details.

Thank you all for the quick reports!

**Comment by medyagh** [comment]
Just FYI this issue seems to break minikube https://github.com/kubernetes/minikube/issues/15634
